src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Options/EditorConfigValueSerializer.cs

```diff
@@ -67,5 +93,55 @@ public static EditorConfigValueSerializer<CodeStyleOption2<bool>> CodeStyle(Code
         public static EditorConfigValueSerializer<CodeStyleOption2<string>> CodeStyle(CodeStyleOption2<string> defaultValue)
             => new(parseValue: str => CodeStyleHelpers.TryParseStringEditorConfigCodeStyleOption(str, defaultValue, out var result) ? result : new Optional<CodeStyleOption2<string>>(),
                    serializeValue: value => value.Value.ToLowerInvariant() + CodeStyleHelpers.GetEditorConfigStringNotificationPart(value, defaultValue));
+
+        public static EditorConfigValueSerializer<T> CreateSerializerForEnum<T>() where T : struct, Enum
+            => new(
+                parseValue: str => TryParseEnum<T>(str, out var result) ? new Optional<T>(result) : new Optional<T>(),
+                serializeValue: value => value.ToString());
+
+        public static EditorConfigValueSerializer<T?> CreateSerializerForNullableEnum<T>() where T : struct, Enum
+        {
+            return new EditorConfigValueSerializer<T?>(
+                parseValue: ParseValueForNullableEnum,
+                serializeValue: value => value == null ? "null" : value.Value.ToString());
+
+            static Optional<T?> ParseValueForNullableEnum(string str)
+            {
+                if (str.Equals("null", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    return new Optional<T?>(null);
+                }
+
+                if (TryParseEnum<T>(str, out var parsedValue))
+                {
+                    return new Optional<T?>(parsedValue);
+                }
+
+                return new Optional<T?>();
+            }
+        }
+
+        private static bool TryParseEnum<T>(string str, out T result) where T : struct, Enum
+        {
+            result = default;
+            // Block any int value.
+            if (int.TryParse(str, out _))
+            {
+                return false;
+            }
+
+            // Enum.TryParse has a strange behavior.
+            // > enum F { A = 1, B = 2, C = 3 }
+            // > Enum.TryParse<F>("A, B", out var x)
+            // > true
+            // > x C
+            // block this case.
```

Comment:
The comment could just say that Enum.TryParses parses every enum as flags enum, but we do not want multiple values to be specified for enums that are not flags.